src/Servers.jl

```diff
@@ -69,15 +69,20 @@ function check_rate_limit(tcp, rate_limit::Rational{Int})
 end
 
 "Convenience object for passing around server details"
-struct Server{S <: Union{SSLConfig, Nothing}, I <: Base.IOServer}
+struct Server{S <: Union{SSLConfig, Nothing}, I <: Base.IOServer, F <: Union{Function, Vector{Function}, Nothing}}
     ssl::S
     server::I
     hostname::String
     hostport::String
+    on_shutdown::F
```

Comment:
It doesn't seem critical to parametrize on this so perhaps just
```suggestion
    on_shutdown::Any
```
would be just as good?

Reply:
Yep, makes sense!